pkg/inventory/inventory-client.go

```diff
@@ -79,79 +71,176 @@ func NewInventoryClient(factory util.Factory) (*ClusterInventoryClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	builder := factory.NewBuilder()
+	builderFunc := factory.NewBuilder
 	clusterInventoryClient := ClusterInventoryClient{
-		builder:                   builder,
-		mapper:                    mapper,
-		validator:                 validator,
-		pastInventoryObjects:      []*resource.Info{},
-		retrievedInventoryObjects: false,
+		builderFunc: builderFunc,
+		mapper:      mapper,
+		validator:   validator,
+		clientFunc:  factory.UnstructuredClientForMapping,
+		dryRun:      dryRun,
+		localInv:    nil,
+		clusterInv:  nil,
 	}
 	return &clusterInventoryClient, nil
 }
 
-// GetStoredObjRefs returns the set of previously applied objects as ObjMetadata,
-// or an error if one occurred. This set of previously applied object references
-// is stored in the inventory objects living in the cluster. Uses information
-// from the current inventory object to find the previously applied inventory
-// objects.
-func (cic *ClusterInventoryClient) GetStoredObjRefs(currentInv *resource.Info) ([]object.ObjMetadata, error) {
-	prevInventories, err := cic.GetPreviousInventoryObjects(currentInv)
+func (cic *ClusterInventoryClient) Init(localInventory *resource.Info) {
+	cic.localInv = localInventory
+}
+
+// Merge stores the union of the passed objects with the objects currently
+// stored in the cluster inventory object. Retrieves and caches the cluster
+// inventory object. Returns the set differrence of the cluster inventory
+// objects and the currently applied objects. This is the set of objects
+// to prune. Creates the initial cluster inventory object storing the passed
+// objects if an inventory object does not exist. Returns an error if one
+// occurred.
+func (cic *ClusterInventoryClient) Merge(objs []object.ObjMetadata) ([]object.ObjMetadata, error) {
+	pruneIds := []object.ObjMetadata{}
+	clusterInv, err := cic.getClusterInventoryInfo()
 	if err != nil {
-		return nil, err
+		return pruneIds, err
 	}
-	return UnionPastObjs(prevInventories)
+	if clusterInv == nil {
+		// Initial create, we must add a unique suffix to the name. Otherwise,
+		// we'll get collisions. The other way to implement this is a unique
+		// suffix appended to name on "init" command.
+		objsHashStr, err := object.Hash(objs)
```

Comment:
Why would we need to add a suffix in this case? We no longer create more than one inventory object, so it seems like it should be ok to just just the name and namespace from the inventory template directly.

Reply:
Removed here. I will add the suffix in a separate PR in the init to generate a unique inventory name that does not collide with other inventory objects.